mitiq/observable/tests/test_observable.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import cirq
+
+from mitiq.observable import PauliString, Observable
+from mitiq.utils import _equal
+
+
+# Basis rotations to measure Pauli X and Y.
+xrotation = cirq.SingleQubitCliffordGate.Y_nsqrt
+yrotation = cirq.SingleQubitCliffordGate.X_sqrt
+
+# Matrices.
+imat = np.identity(2)
+xmat = cirq.unitary(cirq.X)
+zmat = cirq.unitary(cirq.Z)
+
+
+def test_observable():
+    pauli1 = PauliString(spec="XI", coeff=-1.0)
+    pauli2 = PauliString(spec="IZ", coeff=2.0)
+    obs = Observable(pauli1, pauli2)
+
+    assert obs.nqubits == 2
+    assert obs.qubit_indices == [0, 1]
+    assert obs.nterms == 2
+
+    imat = np.identity(2)
+    xmat = cirq.unitary(cirq.X)
+    zmat = cirq.unitary(cirq.Z)
+    correct_matrix = -1.0 * np.kron(xmat, imat) + 2.0 * np.kron(imat, zmat)
+    assert np.allclose(obs.matrix(), correct_matrix)
+
+
+def test_observable_partition_one_set():
+    pauli1 = PauliString(spec="ZI")
+    pauli2 = PauliString(spec="IZ")
+    pauli3 = PauliString(spec="ZZ")
+    obs = Observable(pauli1, pauli2, pauli3)
+    assert obs.nterms == 3
+
+    sets = obs.partition()
+    # expected = {frozenset([pauli1, pauli2, pauli3])}
+
+    assert len(sets) == 1
+    # assert sets == expected
```

Comment:
are there problems here?

Reply:
Yes :) Thanks for reminding me to investigate

Reply:
Outdated, see new test.